tests/fixtures/synthetic_hf_state.py

```diff
@@ -45,9 +47,8 @@ def make_dummy_lm(model_name: str, max_position_embeddings: int, tokenizer):
 
 
 def make_synthetic_dataloader(dataset_config: dict):
-    """
-    creates a dataloader for synthetic sequence data
-    """
+    """creates a dataloader for synthetic sequence data."""
+    pytest.importorskip('transformers')
```

Comment:
@hanlint when we put `importtoskip` inside the functions in #992 it skipped the tests in `test_synthetic_hf_state.py` - we may need to come up with a new work around to avoid skipping. currently in this location my tests are skipped.

Reply:
What should the expected behavior be? For jenkins, we always install `.[all]`, and therefore test everything.

Reply:
We need the importskips, as we have a Jenkins install that tests on just a dev install (this check was fixed in #1163)